eng/common/scripts/add-pullrequest-reviewers.ps1

```diff
@@ -0,0 +1,95 @@
+param(
+    [Parameter(Mandatory = $true)]
+    $RepoOwner,
+
+    [Parameter(Mandatory = $true)]
+    $RepoName,
+
+    [Parameter(Mandatory = $false)]
+    $GitHubUsers = "",
+
+    [Parameter(Mandatory = $false)]
+    $GitHubTeams = "",
+
+    [Parameter(Mandatory = $true)]
+    $PRNumber,
+  
+    [Parameter(Mandatory = $true)]
+    $AuthToken
+)
+
+function CombineArrays($ExistingIds, $NewIds) {
+  if ($ExistingIds -and $ExistingIds.Length > 0) {
+    $results = $ExistingIds.Clone()
+  }
+  else {
+    $results = @()
+  }
+
+  foreach ($id in $NewIds) {
+    if (-not $results.Contains($id)) {
+      $results += $id
+    }
+  }
+
+  return ,$results
+}
+
+# at least one of these needs to be populated
+if (-not $GitHubUsers -and -not $GitHubTeams) {
+  Write-Host "No user provided for addition, exiting."
+  exit 0
+}
+
+$userAdditions = @($GitHubUsers.Split(",") | % { $_.Trim() } | ? { return $_ })
+$teamAdditions = @($GitHubTeams.Split(",") | % { $_.Trim() } | ? { return $_ })
+
+$headers = @{
+  Authorization = "bearer $AuthToken"
+}
+$uri = "https://api.github.com/repos/$RepoOwner/$RepoName/pulls/$PRNumber/requested_reviewers"
+
+try {
+  $resp = Invoke-RestMethod -Headers $headers $uri -MaximumRetryCount 3
+}
+catch {
+  Write-Error "Invoke-RestMethod [$uri] failed with exception:`n$_"
+  exit 1
+}
+
+# the response object takes this form: https://developer.github.com/v3/pulls/review_requests/#response-1
+# before we can push a new reviewer, we need to pull the simple Ids out of the complex objects that came back in the response
+$userReviewers = @($resp.users | % { return $_.login })
+$teamReviewers = @($resp.teams | % { return $_.slug })
+
+$modifiedUserReviewers = CombineArrays -ExistingIds $userReviewers -NewIds $userAdditions
```

Comment:
Out of curiosity is there any reason this wouldn't work? 

```
if (!$usersReviewers) { $userReviewers = @() }
$usersReviews += ($userAdditions | ? { !$usersReviews.Contains($_) })
```

Reply:
I guess if you wanted to detect if there were any new additions you could also store this in its own array and check it below instead of explicitly calling CompareObject.